Quests/Kanto/RainbowBadgeQuest.lua

```diff
@@ -0,0 +1,185 @@
+		return moveToMap("Route 7")
+	elseif isNpcOnCell(54,14) then --Item: Great Ball
+		return talkToNpcOnCell(54,14)
+	elseif isNpcOnCell(46,49) then --NPC: Rocket Guy
+		return talkToNpcOnCell(46,49)
+	elseif isNpcOnCell(58,51) then --NPC: Rocket Guy
+		return talkToNpcOnCell(58,51)
+	elseif isNpcOnCell(50,55) then --Item: 5x Ultra Balls
+		return talkToNpcOnCell(50,55)
+	elseif not game.hasPokemonWithMove("Cut") then
+		if pokemonId < getTeamSize() then					
+			useItemOnPokemon("HM01 - Cut", self.pokemonId)
+			log("Pokemon: " .. self.pokemonId .. " Try Learning: HM01 - Cut")
+			self.pokemonId = self.pokemonId + 1
+		else
+			fatal("No pokemon in this team can learn - Cut")
+		end
+	elseif not hasItem("Rainbow Badge") then
+		return moveToMap("CeladonGym")
+	elseif isNpcOnCell(14,42) then --NPC: Remove the Guards
+		return talkToNpcOnCell(14,42)
+	elseif not hasItem("Fresh Water") then -- Buy Fresh Water for Future Quest (Saffron Guard)
+		return moveToMap("Celadon Mart 1")
+	else
+		return moveToMap("Route 7")
+	end
+end
+
+function RainbowBadgeQuest:PokecenterCeladon()
+	self:pokecenter("Celadon City")
+end
+
+function RainbowBadgeQuest:Route7()
+	if self:needPokecenter() or not game.isTeamFullyHealed() or not self.registeredPokecenter == "Pokecenter Celadon" then
+		return moveToMap("Celadon City")
+	elseif hasItem("Rainbow Badge") and hasItem("Fresh Water") then
+		return moveToMap("Underground House 3")
+	elseif not self:isTrainingOver() and not hasItem("Rainbow Badge") then
+		if not game.inRectangle(12,8,21,21) then
+			return moveToCell(17,17)
+		else
+			return moveToGrass()
+		end
+	else
+		return moveToMap("Celadon City")
+	end
+end
+
+function RainbowBadgeQuest:CeladonGym()
+	if self:needPokecenter() or not game.isTeamFullyHealed() or not self.registeredPokecenter == "Pokecenter Celadon" or not self:isTrainingOver() then
+		return moveToMap("Celadon City")
+	elseif not hasItem("Rainbow Badge") then
+		talkToNpcOnCell(8,4) -- Erika
+	else
+		return moveToMap("Celadon City")
+	end
+end
+
+function RainbowBadgeQuest:CeladonMart1()
+	if not hasItem("Fresh Water") then
+		return moveToMap("Celadon Mart Elevator")
+	else
+		return moveToMap("Celadon City")
+	end
+end
+
+function RainbowBadgeQuest:CeladonMartElevator()
+	if not hasItem("Fresh Water") then
+		if not dialogs.martElevatorFloor5.state then
+			pushDialogAnswer(5)
+			return talkToNpcOnCell(1,1)
+		else
+			dialogs.martElevatorFloor5.state = false
+			return moveToCell(2,5)
+		end
+	else
+		if not dialogs.martElevatorFloor1.state then
+			pushDialogAnswer(1)
+			return talkToNpcOnCell(1,1)
+		else
+			dialogs.martElevatorFloor1.state = false
+			return moveToCell(2,5)
+		end
+	end
+end
+
+function RainbowBadgeQuest:CeladonMart5()
+	if not hasItem("Fresh Water") then
+		return moveToMap("Celadon Mart 6")
+	else
+		return moveToMap("Celadon Mart Elevator")
+	end
+end
+
+function RainbowBadgeQuest:CeladonMart6()
+	if not hasItem("Fresh Water") then
```

Comment:
`if not hasItem("Fresh Water") then`

is not correct,
Change it to

`if not hasItem("Lemonade") then`

else it will get stuck in an loop trying to buy Lemonade all the time.